gremlin-test/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/step/OrderabilityTest.java

```diff
@@ -0,0 +1,537 @@
+                "lop",      // vid = 3, val = "lop"
+                "java",     // vid = 5, val = "java"
+                "ripple"    // vid = 5, val = "ripple"
+        ), traversal);
+    }
+
+    /**
+     * Order desc by vertex: v[5], v[3]
+     *
+     * Note to graph providers: if your graph does not support user-assigned vertex ids you may need to
+     * skip this test.
+     */
+    @Test
+    @LoadGraphWith(MODERN)
+    public void g_V_out_out_order_byXdescX() {
+        final Traversal traversal = get_g_V_out_out_order_byXdescX();
+        printTraversalForm(traversal);
+        checkOrderedResults(Arrays.asList(
+                convertToVertex("ripple"),     // vid = 5
+                convertToVertex("lop")         // vid = 3
+        ), traversal);
+    }
+
+    /**
+     * Order desc by vertex in path: v[5], v[3]
+     *
+     * Note to graph providers: if your graph does not support user-assigned vertex ids you may need to
+     * skip this test.
+     */
+    @Test
+    @LoadGraphWith(MODERN)
+    public void g_V_out_out_asXheadX_path_order_byXdescX_selectXheadX() {
+        final Traversal traversal = get_g_V_out_out_asXheadX_path_order_byXdescX_selectXheadX();
+        printTraversalForm(traversal);
+        checkOrderedResults(Arrays.asList(
+                convertToVertex("ripple"),     // vid = 5
+                convertToVertex("lop")         // vid = 3
+        ), traversal);
+    }
+
+    /**
+     * Order desc by edge: e[11], v[e10]
+     *
+     * Note to graph providers: if your graph does not support user-assigned edge ids you may need to
+     * skip this test.
+     */
+    @Test
+    @LoadGraphWith(MODERN)
+    public void g_V_out_outE_order_byXdescX() {
+        final Traversal traversal = get_g_V_out_outE_order_byXdescX();
+        printTraversalForm(traversal);
+        checkOrderedResults(Arrays.asList(
+                convertToEdge("josh", "created", "lop"),        // eid = 11
+                convertToEdge("josh", "created", "ripple")      // eid = 10
+        ), traversal);
+    }
+
+    /**
+     * Order desc by edge in path: e[11], e[10]
+     *
+     * Note to graph providers: if your graph does not support user-assigned edge ids you may need to
+     * skip this test.
+     */
+    @Test
+    @LoadGraphWith(MODERN)
+    public void g_V_out_outE_asXheadX_path_order_byXdescX_selectXheadX() {
+        final Traversal traversal = get_g_V_out_outE_asXheadX_path_order_byXdescX_selectXheadX();
+        printTraversalForm(traversal);
+        checkOrderedResults(Arrays.asList(
+                convertToEdge("josh", "created", "lop"),        // eid = 11
+                convertToEdge("josh", "created", "ripple")      // eid = 10
+        ), traversal);
+    }
+
+    /**
+     * Order desc by vertex and then vertex property id in path.
+     *
+     * Note to graph providers: if your graph does not support user-assigned vertex ids and vertex property ids you 
```

Comment:
you can already check the feature support in the tests using `@FeatureRequirement` annotation.

```
@FeatureRequirement(featureClass = ElementFeatures.class, feature = FEATURE_USER_SUPPLIED_IDS)
```

Reply:
done